CHANGELOG.md

```diff
@@ -119,7 +125,37 @@ Changelog
   are passed to the QVM by `WavefunctionSimulator.run_and_measure()` and `WavefunctionSimulator.wavefunction()`.
   
 - `noise.estimate_assignment_probs()` now accepts a `QuantumComputer` instead of `QVMConnection`.
-  
+
+- `QAM` and its subclasses (such as `QPU` and `QVM`) do not store any information specific to the state
+  of execution requests, and thus are safe to be used concurrently by different requests. `QAM.run`
+  is now composed of two intermediate calls:
+  -  `QAM.execute` starts execution of the provided executable, returning an opaque handle.
+  -  `QAM.get_result` uses the opaque handle returned by `execute` to retrieve the result values.
+
+  These new calls can be used to enqueue multiple programs for execution prior to retrieving
+  results for any of them. Note that this new pattern means that `QAM.load`, `QAM.reset`, and
+  `QAM.wait` no longer exist.
+
+- `QAM.run` no longer accepts a `memory_map` argument. Memory values must be written onto
+  executable directly with `Program.write_memory()` and `EncryptedProgram.write_memory()` instead. 
+
+- `QuantumComputer`, `QAM`, `QPU`, and `QVM` are now safe to share across threads and processes,
+  as they no longer store request-related state.
+
+- `PyQVM.execute` has been renamed to `PyQVM.execute_once` to execute a single program from start
+  to finish within the context of the existing `PyQVM` state. `PyQVM` is the only stateful `QAM`.
+  `PyQVM.execute` now implements `QAM.execute` and resets the `PyQVM` state prior to program execution.
+
+- `QuantumComputer.experiment` has been renamed to `QuantumComputer.run_experiment`.
+
+- Results returned from execution are now referred to as `readout_data` rather than `memory`, reflecting the reality
+  that the memory of the QAM is not currently exposed to the user. The exception to this rule is the stateful `PyQVM`,
+  whose state is maintained within the pyQuil process and whose memory _may truly be inspected._ For that,
+  `PyQVM.read_memory` remains available.
+
+- `QuantumComputer.run` now returns a `QAMExecutionResult` rather than the readout data from the `ro` readout
```

Comment:
Looks like the code example in the README is also in need of updating, regarding this change.

Reply:
good catch!